deltacat/compute/compactor/model/delta_file_envelope.py

```diff
@@ -42,11 +49,14 @@ def of(
             raise ValueError("Missing Delta file envelope delta type.")
         if table is None:
             raise ValueError("Missing Delta file envelope table.")
+        if table_storage_strategy is None:
+            table_storage_strategy = LocalTableNOOPStorageStrategy()
```

Comment:
I am not sure what impact this would have on performance. Ray is good at serializing dict but i am not sure about ABC.

Reply:
I'll investigate this

Reply:
```
from deltacat.compute.compactor import (
    DeltaFileEnvelope,
)
class DeltaFileEnvelopeNoTableStorageStrategy(dict):
    @staticmethod
    def of(
        stream_position: int,
        delta_type: DeltaType,
        table: LocalTable,
        file_index: int = None,
        is_src_delta: np.bool_ = True,
        file_record_count: Optional[int] = None,
    ) -> DeltaFileEnvelope:
        """
        Static factory builder for a Delta File Envelope
        `
        Args:
            stream_position: Stream position of a delta.
            delta_type: A delta type.
            table: The table object that represents the delta file.
            file_index: Manifest file index number of a delta.
            is_src_delta: True if this Delta File Locator is
                pointing to a file from the uncompacted source table, False if
                this Locator is pointing to a file in the compacted destination
                table.
            table_storage_strategy: The way the table object is stored in the delta file envelope
        Returns:
            A delta file envelope.

        """
        if stream_position is None:
            raise ValueError("Missing delta file envelope stream position.")
        if delta_type is None:
            raise ValueError("Missing Delta file envelope delta type.")
        if table is None:
            raise ValueError("Missing Delta file envelope table.")
        delta_file_envelope = DeltaFileEnvelope()
        delta_file_envelope["streamPosition"] = stream_position
        delta_file_envelope["fileIndex"] = file_index
        delta_file_envelope["deltaType"] = delta_type.value
        delta_file_envelope["table"] = table
        delta_file_envelope["is_src_delta"] = is_src_delta
        delta_file_envelope["file_record_count"] = file_record_count
        return delta_file_envelope

    @property
    def stream_position(self) -> int:
        return self["streamPosition"]

    @property
    def file_index(self) -> int:
        return self["fileIndex"]

    @property
    def delta_type(self) -> DeltaType:
        return DeltaType(self["deltaType"])

    @property
    def table(self) -> LocalTable:
        return self["table"]

    @property
    def is_src_delta(self) -> np.bool_:
        return self["is_src_delta"]

    @property
    def file_record_count(self) -> int:
        return self["file_record_count"]

    @property
    def table_size_bytes(self) -> int:
        if isinstance(self.table, pa.Table):
            return self.table.nbytes
        else:
            raise ValueError(
                f"Table type: {type(self.table)} not for supported for size method."
            )

    @property
    def table_num_rows(self) -> int:
        return len(self.table)

df_envelopes_with_no_table_storage_strategy = [
    DeltaFileEnvelopeNoTableStorageStrategy.of(100, DeltaType.UPSERT, file_index=1, is_src_delta=True, file_record_count=1000, table=pa.table({"col1": [1, 2, 3]})), 
    DeltaFileEnvelopeNoTableStorageStrategy.of(200, DeltaType.UPSERT,  file_index=2, is_src_delta=True, file_record_count=2000,  table=pa.table({"col1": [1, 2, 3]})), 
    DeltaFileEnvelopeNoTableStorageStrategy.of(400, DeltaType.UPSERT,  file_index=3, is_src_delta=False, file_record_count=3000,  table=pa.table({"col1": [1, 2, 3]})),
    DeltaFileEnvelopeNoTableStorageStrategy.of(700, DeltaType.UPSERT,  file_index=4, is_src_delta=False, file_record_count=4000,  table=pa.table({"col1": [1, 2, 3]})),
]
df_envelopes_with_table_storage_strategy = [
    DeltaFileEnvelope.of(100, DeltaType.UPSERT, file_index=1, is_src_delta=True, file_record_count=1000, table=pa.table({"col1": [1, 2, 3]})),
    DeltaFileEnvelope.of(200, DeltaType.UPSERT,  file_index=2, is_src_delta=True, file_record_count=2000,  table=pa.table({"col1": [1, 2, 3]})),
    DeltaFileEnvelope.of(400, DeltaType.UPSERT,  file_index=3, is_src_delta=False, file_record_count=3000,  table=pa.table({"col1": [1, 2, 3]})),
    DeltaFileEnvelope.of(700, DeltaType.UPSERT,  file_index=4, is_src_delta=False, file_record_count=4000,  table=pa.table({"col1": [1, 2, 3]})),
]
dumped_no_table_storage_strategy = cloudpickle.dumps(df_envelopes_with_no_table_storage_strategy)
print(len(dumped_no_table_storage_strategy)) # 1010
dumped_w_table_storage_strategy = cloudpickle.dumps(df_envelopes_with_table_storage_strategy)
print(len(dumped_w_table_storage_strategy)) # 1145
```

Got an approximately 13% increase in size. Subclassing of a dictionary had no impact

Reply:
For 10000 elements it's around a 3% increase
```
df_envelopes_with_no_table_storage_strategy = [
    DeltaFileEnvelopeNoTableStorageStrategy.of(100, DeltaType.UPSERT, file_index=1, is_src_delta=True, file_record_count=1000, table=pa.table({"col1": [1, 2, 3]}))
    for _ in range(10000)
]
df_envelopes_with_table_storage_strategy = [
    DeltaFileEnvelope.of(100, DeltaType.UPSERT, file_index=1, is_src_delta=True, file_record_count=1000, table=pa.table({"col1": [1, 2, 3]}))
    for _ in range(10000)
]
dumped_no_table_storage_strategy = cloudpickle.dumps(df_envelopes_with_no_table_storage_strategy)
print(len(dumped_no_table_storage_strategy)) # 1780557
dumped_w_table_storage_strategy = cloudpickle.dumps(df_envelopes_with_table_storage_strategy)
print(len(dumped_w_table_storage_strategy)) # 1850673

```

Reply:
We can avoid the 3% increase by not having NoOP strategy.

Reply:
Done